tests/unit/string_util_test.cpp

```diff
@@ -0,0 +1,62 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <gtest/gtest.h>
+
+using namespace testing;
+
+extern "C" {
+   #include <tilck/common/string_util.h>
+   char *tilck_strstr(const char *haystack, const char *needle);
```

Comment:
Of course declaring here the `tilck_*` functions works, but it's not safe. That's why I said that you have to put those in the string_util.h header.

Why it's not safe? Because if for any reason somebody changes the signature of one of those functions in the kernel without fixing these ones, no warnings or errors will be shown. That's simply because these declarations are not visible while compiling their bodies. It's like declaring a function twice in two separate C files, instead of using a header. **It's very dangerous.** [1]

Therefore, add those in `string_util.h` and add the short names (e.g. strcmp etc.) under `#if !defined(TESTING) && !defined(USERMODE_APP)`. At the beginning of that header file there is already such a block. Just append them at the bottom of it and add the `tilck_*` ones outside of the `#ifdef`.  Makes sense?

----------------------
[1] It's true that sometimes I might have told you to just declare the kernel function in the unit test. That applies for STATIC functions that are not declared in any header. And it's still dangerous for the same reasons. I'm starting to think that we should introduce special test-only headers with all of those declarations, just to get that extra compile-time safety.

Reply:
Sounds good. Sorry for the extra hassle, but i want to make sure i understand the reasoning behind some choices.